llvm/lib/Transforms/Scalar/Reassociate.cpp

```diff
@@ -1225,7 +1234,7 @@ Value *ReassociatePass::RemoveFactorFromExpression(Value *V, Value *Factor) {
     RedoInsts.insert(BO);
     V = Factors[0].Op;
   } else {
-    RewriteExprTree(BO, Factors);
+    RewriteExprTree(BO, Factors, HasNUW);
```

Comment:
Could there be a problem if NeedsNegate is set?

Reply:
I don't think it is an issue.
The transform `-C *nuw X1 *nuw X2 *nuw X3` --> `C * -(X1 *nuw X2 *nuw X3)` is correct.